substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jfr/events/NativeMemoryUsagePeakEvent.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2024, 2024, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2024, 2024, Red Hat Inc. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package com.oracle.svm.core.jfr.events;
+
+import jdk.jfr.Category;
+import jdk.jfr.Description;
+import jdk.jfr.Event;
+import jdk.jfr.Experimental;
+import jdk.jfr.Label;
+import jdk.jfr.Name;
+import jdk.jfr.StackTrace;
+
+/** Similar to the JFR event jdk.NativeMemoryUsage, except that it tracks the peak usage. */
+@Experimental
+@Name("jdk.NativeMemoryUsagePeak")
+@Label("Native Memory Usage Peak")
+@Description("Native memory peak usage for a given memory type in the JVM.")
+@Category({"Java Virtual Machine", "Memory"})
+@StackTrace(false)
+public class NativeMemoryUsagePeakEvent extends Event {
```

Comment:
All of the other events with the "jdk" prefix have been ported over from OpenJDK. Maybe there should be some way of differentiating these NMT events. If we want to continue using the "jdk" prefix, do you think it would be a good idea to include a note in the event description indicating these events are built-in specific to Native Image?

Reply:
We've also marked the new events as experimental for now, that's how I would currently look for them. But yes, a short note like `(GraalVM Native Image only)` or so in the description sounds reasonable to me.